### ISSUE TITLE
Implement OTP Verification for Signup and Password Reset

### ISSUE THREAD

**Issue by HarshitShukla-dev** [AkshitLakhera/PenCraft-Full-Stack-Blogging-Application#18]
Hi,
I would like to propose adding an OTP verification feature during the signup process. Additionally, this functionality will also be implemented for the "Forgot Password" feature to enhance security.

Proposed changes:
1. Implement OTP verification for user signup.
2. Include OTP verification for the password reset process.
3. Ensure that the OTP can be sent via email, based on the existing user information.

This enhancement will significantly improve the security of user accounts. Could you please assign this issue to me?

Thank you!

**Comment by AkshitLakhera** [comment]
Hi @HarshitShukla-dev ,
Thank you for bringing up this enhancement. 
I've assigned the issue to you. Please go ahead with the implementation, and do let me know if you face any difficulties along the way. I'm happy to assist if needed.

**Comment by kushagra0304** [comment]
Hey @HarshitShukla-dev,
Issue #20 is assigned to me and it works in conjunction with your issue #18.
Can we work together to fix the issues?

**Comment by AkshitLakhera** [comment]
See guys @kushagra0304  @HarshitShukla-dev  , there is  one other guy who also have suggested  to add  forgot password feature.I want you all guys  work together and complete this whole thing.I will make this level 3 issue and  add all three of you in this. Okay  ?

**Comment by Sourabh782** [comment]
@HarshitShukla-dev @kushagra0304 my discord tag is Sourabh782#4309  and display name is hero123 in gssoc24 channel, tag me so we can rotk together

**Comment by Sourabh782** [comment]
@AkshitLakhera, actually i and @HarshitShukla-dev  recently connected for the project working flow, and decided to start working on it. So can we start contributing to it?

**Comment by HarshitShukla-dev** [comment]
> See guys @kushagra0304 @HarshitShukla-dev , there is one other guy who also have suggested to add forgot password feature.I want you all guys work together and complete this whole thing.I will make this level 3 issue and add all three of you in this. Okay ?

Yeah, it works

**Comment by AkshitLakhera** [comment]
> @AkshitLakhera, actually i and @HarshitShukla-dev recently connected for the project working flow, and decided to start working on it. So can we start contributing to it?

Yes you both can start if @kushagra0304  you be interested I will add him too.

**Comment by AkshitLakhera** [comment]
@HarshitShukla-dev  @Sourabh782   any progress update guys ?

**Comment by Sourabh782** [comment]
yes @AkshitLakhera, currently working on frontend part,

![Screenshot 2024-10-04 172526](https://github.com/user-attachments/assets/21b45857-9b88-496b-aab1-66ecf733cffc)

Email taking page done, now designing otp page to take otp input and new password

**Comment by AkshitLakhera** [comment]
> yes @AkshitLakhera, currently working on frontend part,
> 
> ![Screenshot 2024-10-04 172526](https://private-user-images.githubusercontent.com/103349890/373628521-21b45857-9b88-496b-aab1-66ecf733cffc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgwNDQ5MTUsIm5iZiI6MTcyODA0NDYxNSwicGF0aCI6Ii8xMDMzNDk4OTAvMzczNjI4NTIxLTIxYjQ1ODU3LTliODgtNDk2Yi1hYWIxLTY2ZWNmNzMzY2ZmYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwNFQxMjIzMzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNTcyMjhlZTg5ZTNkZjNlZWY0OTY5NzhmZWMxY2YwNzdmMWE3OWNlZDNhMDlkYTYwM2JjZDNkMDZkNTg1YTRkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.AflO86l_kyAVxV6l6UBm8Oe3rXjizggS0V-2GScM858)
> 
> Email taking page done, now designing otp page to take otp input and new password

Good going guys, which service are you  using for otp sending ?

**Comment by Sourabh782** [comment]
@AkshitLakhera hii we haven't figured out it yet, but I have experience with resend mail. Do you have any good ideas about which services should we use ?

**Comment by AkshitLakhera** [comment]
> @AkshitLakhera we haven't figured out it yet, but I have experience with resend mail. Do you have any good ideas about which services should we use ?

Yeah ,go with resend it is  best as far as I know.

**Comment by Sourabh782** [comment]
> > @AkshitLakhera we haven't figured out it yet, but I have experience with resend mail. Do you have any good ideas about which services should we use ?
> 
> Yeah ,go with resend it is best as far as I know.

Sure, and actually I have some exam on weekend so till next week will complete it

**Comment by Sourabh782** [comment]
@AkshitLakhera Hii can you verify frontend part?
![Screenshot 2024-10-05 182454](https://github.com/user-attachments/assets/27a174d8-48f5-493d-9a6a-eb82d323618a)
![Screenshot 2024-10-05 182446](https://github.com/user-attachments/assets/e0b7b196-5679-4eae-96ca-382b5f4ae92a)
![Screenshot 2024-10-05 182435](https://github.com/user-attachments/assets/a602a350-2135-4b51-9065-95f05d7ff9fb)
![Screenshot 2024-10-05 182421](https://github.com/user-attachments/assets/2c3a76ca-ebce-43fe-80a0-09fa9f416e74)

after reset password, user will be sent to signin page

**Comment by Sourabh782** [comment]
@HarshitShukla-dev can you message me on discord?

**Comment by AkshitLakhera** [comment]
@Sourabh782  are things are working fine in your end? Are you able to signin /up ?

**Comment by Sourabh782** [comment]
@AkshitLakhera actually we decided to split tasks and i developed frontend. but i havent heard from him since 2nd

**Comment by AkshitLakhera** [comment]
> @AkshitLakhera actually we decided to split tasks and i developed frontend. but i havent heard from him since 2nd

you complete from your end , if he will not show up I will assign only to you

**Comment by AkshitLakhera** [comment]
@Sourabh782   what's your discord username ?

**Comment by Sourabh782** [comment]
> @Sourabh782 what's your discord username ?

sourabh782

**Comment by Sourabh782** [comment]
> > @AkshitLakhera actually we decided to split tasks and i developed frontend. but i havent heard from him since 2nd
> 
> you complete from your end , if he will not show up I will assign only to you

My part is ready just have to add backend calls and its good to go

**Comment by AkshitLakhera** [comment]
@Sourabh782   come in discord , I want to tell something to you .
